### DISCUSSION TITLE
Setting max select values

### DISCUSSION THREAD

**Discussion by gaurishhs** [csandman/chakra-react-select#158] [Q&A]
How can i set max values that can be selected?

**Comment by csandman** [comment]
I'm not sure exactly what you mean, do you mean a max number of options that can be selected in a multi-select?

**Comment by gaurishhs** [comment]
Yes, I meant that

**Comment by csandman** [comment]
This is not something that would be handled by this library directly, you'd have to add some custom logic to enable this functionality. With a Google search, I see plenty of examples from the original library `react-select` that could help you if you just replace it with `chakra-react-select`:

- https://stackoverflow.com/questions/55119120/how-to-set-max-number-of-items-that-can-be-selected-in-react-select
- https://stackoverflow.com/questions/69388245/how-to-define-a-max-limit-of-options-in-react-select
- https://github.com/JedWatson/react-select/issues/1341

Here's one way you could do it by replacing the list of options with a single disabled option:

```js
import React, { useState } from "react";
import { Container, FormControl, FormLabel } from "@chakra-ui/react";
import { Select } from "chakra-react-select";
import { stateOptions } from "./data/data";

const MAX_SELECTED = 4;

const maxSelectedOptions = [
  {
    isDisabled: true,
    value: "max",
    label: "You have selected the max number of states"
  }
];

const Example = () => {
  const [selectedStates, setSelectedStates] = useState([]);

  const isMaxed = selectedStates.length === MAX_SELECTED;

  return (
    <Container mb={16}>
      <FormControl p={4}>
        <FormLabel>
          Select up to {MAX_SELECTED} states you want to visit
        </FormLabel>
        <Select
          isMulti
          name="states"
          options={isMaxed ? maxSelectedOptions : stateOptions}
          value={selectedStates}
          onChange={setSelectedStates}
          placeholder="Select some states..."
          closeMenuOnSelect={false}
        />
      </FormControl>
    </Container>
  );
};

export default Example;
```

CodeSandbox: https://codesandbox.io/s/chakra-react-select-max-selected-options-5u36gx?file=/example.js

**Reply by gaurishhs** [reply]
@csandman How would i do this with react-hook-form?

**Reply by csandman** [reply]
@gaurishhs Do you mean using a form validation lib like `yup` or `zod`?  Because the above example could apply to `react-hook-form` as well.
